app/src/main/java/homer/model/lock/SimpleLock.java

```diff
@@ -0,0 +1,39 @@
+package homer.model.lock;
+
+import homer.DeviceInfoImpl;
+import homer.api.DeviceIdImpl;
+import homer.api.DeviceInfo;
+import homer.api.ToggleableDevice;
+
+/**
+ * Implementation of a {@link ToggleableDevice} representing a lock.
+ */
+public final class SimpleLock implements ToggleableDevice<Boolean> {
+
+    /**
+     * Type of device.
+     */
+    public static final String DEVICE_TYPE = "SimpleLock";
+    private boolean isLocked = false;
+
+    @Override
+    public DeviceInfo getInfo() {
+        return new DeviceInfoImpl(new DeviceIdImpl(), DEVICE_TYPE);
+    }
+
+    @Override
+    public Boolean getState() {
+        return this.isLocked;
+    }
+
+    @Override
+    public boolean isToggled() {
+        return getState();
+    }
+
+    @Override
+    public void toggle() {
+        this.isLocked = !this.isLocked;
```

Comment:
Use xor to toggle the device:
```
this.isLocked = this.isLocked ^= true;
```

Reply:
Not sure what this would improve.

Reply:
Fewer keystrokes and i think it might be a good idea to use
`Boolean isLocked` instead of `boolean isLocked`.
Using XOR is safe in case isLocked is null, using NOT isn't (will throw NullPointerException)
By the way, it's just a "style" thing, doesn't really matters.
Look at my impl of Lights. It's just for the methods to be compliant between them i guess.
@michelececcacci what'd you say? 
NOT or XOR that is the question... /s

Reply:
`Boolean` is the class wrapping a `boolean` field.
`boolean` is the primitive type, it cannot be null.
Our code does not have to be identical, actually that would go against the point of the project

Reply:
Fine then, not going to disagree :)

Reply:
Fine then, not going to disagree :)